.clang-tidy

```diff
@@ -1,42 +1,32 @@
 Checks: "-*,\
 bugprone-*,\
+cert-dcl21-cpp,\
+cert-dcl50-cpp,\
+cert-env33-c,\
+-cert-err34-c,\
+cert-err52-cpp,\
+-cert-err58-cpp,\
+cert-err60-cpp,\
+-cert-flp30-c,\
 -clang-analyzer-security.FloatLoopCounter,\
 google-build-namespaces,\
--google-explicit-constructor,\
+google-explicit-constructor,\
 google-readability-todo,\
 google-runtime-operator,\
+misc-new-delete-overloads,\
+misc-non-copyable-objects,\
+misc-static-assert,\
 misc-suspicious-*,\
+misc-throw-by-value-catch-by-reference,\
 misc-unused-*,\
 modernize-loop-convert,\
 modernize-make-shared,\
 modernize-make-unique,\
```

Comment:
I think we need to enable more of these checks

Reply:
I left in a few disabled ones which are my recommendations to enable next. We have a lot of `cert-err58-cpp` violations, for example, where we construct statics and globals that may throw on construction.

Reply:
`cert-err58-cpp` will be harder to remove but I would recommend enabling all the `modernize-*` checks